tests/fsharp/core/printf/test.fsx

```diff
@@ -384,6 +384,149 @@ module CheckDisplayAttributes8 =
+
+// Check one returning an int and a list
+module CheckDisplayAttributes10 =
+
+    [<StructuredFormatDisplay("{Hello}: {StructuredDisplay}")>]
+    type Foo() = 
+       member internal x.StructuredDisplay = [1;2]
+       member internal x.Hello = "Hello"
+       override x.ToString() = "2"
+
+    test "cenwoiwe10" (lazy(sprintf "%A" (Foo()))) "Hello: [1; 2]"
+
+// Check one returning an int and a string with no spaces
+module CheckDisplayAttributes11 =
+
+    [<StructuredFormatDisplay("{Val}{Hello}")>]
+    type Foo() = 
+       member internal x.Val = 42
+       member internal x.Hello = "Hello"
+
+    test "cenwoiwe11" (lazy(sprintf "%A" (Foo()))) "42Hello"
+
+// Check one with an unmatched opening bracket
+module CheckDisplayAttributes12 =
+
+    [<StructuredFormatDisplay("{Val{Hello}")>]
+    type Foo() = 
+       member internal x.Val = 42
+       member internal x.Hello = "Hello"
+       override x.ToString() = "x"
+
+    // this should produce an error message
+    test "cenwoiwe12" (lazy(sprintf "%A" (Foo()))) "<StructuredFormatDisplay exception: Method 'Test+CheckDisplayAttributes12+Foo.Val{Hello' not found.>"
+
+// Check one with an unmatched closing bracket
+module CheckDisplayAttributes13 =
+
+    [<StructuredFormatDisplay("{Val}Hello}")>]
+    type Foo() = 
+       member internal x.Val = 42
+       member internal x.Hello = "Hello"
+       override x.ToString() = "x"
+
+    test "cenwoiwe13" (lazy(sprintf "%A" (Foo()))) "x"
+
+// Check one with an unmatched trailing open bracket
+module CheckDisplayAttributes14 =
+
+    [<StructuredFormatDisplay("{Val}{Hello")>]
+    type Foo() = 
+       member internal x.Val = 42
+       member internal x.Hello = "Hello"
+       override x.ToString() = "x"
+
+    test "cenwoiwe14" (lazy(sprintf "%A" (Foo()))) "x"
+
+// Check one with unbounded recursion
+module CheckDisplayAttributes15 =
+
+    [<StructuredFormatDisplay("{X} {X}")>]
+    type Foo() = 
+       member internal x.X = Foo()
+
+    test "cenwoiwe15" (lazy(sprintf "%A" (Foo()))) "... ... ... ... ... ... ... ..."
+
+// Check escaped brackets with no other members
+module CheckDisplayAttributes16 =
+
+    [<StructuredFormatDisplay("{\{\}}")>]
+    type Foo() = 
+      member __.``{}`` = "abc"
+
+    test "cenwoiwe16" (lazy(sprintf "%A" (Foo()))) "abc"
+
+// Check escaped brackets with other members
+module CheckDisplayAttributes17 =
+
+    [<StructuredFormatDisplay("{One\} \{Two}")>]
+    type Foo() =
+      member __.``One} {Two`` = "abc"
+      member __.One = 123
+      member __.Two = 456
+
+    test "cenwoiwe17" (lazy(sprintf "%A" (Foo()))) "abc"
+
+// Check escaped brackets with all brackets escaped
+module CheckDisplayAttributes18 =
+
+    [<StructuredFormatDisplay("\{One\} \{Two\}")>]
+    type Foo() =
+      member __.``One} {Two`` = "abc"
+      member __.One = 123
+      member __.Two = 456
+      override x.ToString() = "x"
+
+    test "cenwoiwe18" (lazy(sprintf "%A" (Foo()))) "x"
```

Comment:
This looks like a perfectly good string, with all escaped brackets.  Why should it not print `"{One} {Two}"`?